internal/envoy/regex.go

```diff
@@ -18,13 +18,23 @@ import (
 	"github.com/projectcontour/contour/internal/protobuf"
 )
 
+// DefaultMaxRegexProgramSize is the default value for the Envoy regex
+// max program size tunable. There's no way to really know what a good
+// value for this is, except that the RE2 maintainer thinks that 100
+// is low. As a rule of thumb, each '.*' expression costs about 15
+// units of program size. AFAIK, there's no obvious correlation between
+// regex size and execution time.
+//
+// https://github.com/envoyproxy/envoy/pull/9171#discussion_r351974033
+const DefaultMaxRegexProgramSize = 1000
+
 // SafeRegexMatch retruns a matcher.RegexMatcher for the supplied regex.
 // SafeRegexMatch does not escape regex meta characters.
 func SafeRegexMatch(regex string) *matcher.RegexMatcher {
 	return &matcher.RegexMatcher{
 		EngineType: &matcher.RegexMatcher_GoogleRe2{
 			GoogleRe2: &matcher.RegexMatcher_GoogleRE2{
-				MaxProgramSize: protobuf.UInt32(uint32(len(regex))),
+				MaxProgramSize: protobuf.UInt32(uint32(DefaultMaxRegexProgramSize)),
```

Comment:
nit: you don't need to convert an untyped constant to a uint32, the compiler will do that for you 
```
MaxProgramSize: protobuf.UInt32(DefaultMaxRegexProgramSize)
```

Please don't call this `DefaultMaxRegexProgramSize`. It's not the default, it might be our default, but even then I don't think calling it default is accurate, `maxProgramSize` is fine.

Reply:
Thanks, fixed!